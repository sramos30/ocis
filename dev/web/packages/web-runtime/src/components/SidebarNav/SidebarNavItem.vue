<template>
  <li class="oc-sidebar-nav-item oc-pb-xs oc-px-s" :aria-current="active ? 'page' : null">
    <oc-button
      v-oc-tooltip="toolTip"
      type="router-link"
      appearance="raw"
      :variation="active ? 'inverse' : 'passive'"
      :class="['oc-sidebar-nav-item-link', { active: active }]"
      :to="target"
      :data-nav-id="index"
      :data-nav-name="navName"
    >
      <span class="oc-flex">
        <oc-icon :name="icon" :fill-type="fillType" />
        <span class="oc-ml-m text" :class="{ 'text-invisible': collapsed }" v-text="name" />
      </span>
      <oc-tag v-if="tag" class="oc-py-rm" size="small">{{ tag }}</oc-tag>
      <sidebar-nav-item-highlight :index="index" :active="active" />
    </oc-button>
  </li>
</template>
<script>
import SidebarNavItemHighlight from './SidebarNavItemHighlight.vue'
import get from 'lodash-es/get'

export default {
  components: {
    SidebarNavItemHighlight
  },
  props: {
    name: {
      type: String,
      required: true
    },
    index: {
      type: Number,
      required: true
    },
    active: {
      type: Boolean,
      required: false,
      default: false
    },
    target: {
      type: String,
      required: false,
      default: null
    },
    icon: {
      type: String,
      required: true
    },
    fillType: {
      type: String,
      required: false,
      default: 'fill'
    },
    collapsed: {
      type: Boolean,
      required: false,
      default: false
    },
    tag: {
      type: String,
      required: false,
      default: null
    }
  },
  computed: {
    navName() {
      return get(this.$router?.resolve(this.target), 'route.name')
    },
    toolTip() {
      const value = this.collapsed
        ? this.$gettextInterpolate(this.$gettext('Navigate to %{ pageName } page'), {
            pageName: this.name
          })
        : ''
      return {
        content: value,
        placement: 'right',
        arrow: false
      }
    }
  }
}
</script>

<style lang="scss">
.oc-sidebar-nav-item-link {
  position: relative;
  align-items: center !important;
  display: flex !important;
  justify-content: space-between !important;
  padding: var(--oc-space-small) !important;
  border-radius: 5px;
  white-space: nowrap;
  user-select: none;

  .oc-tag {
    color: var(--oc-color-text-default);
    background-color: var(--oc-color-background-highlight);
  }
  .text {
    opacity: 1;
    transition: all 0s;
    transition-delay: 0.1s;
  }
  .text-invisible {
    opacity: 0 !important;
    transition: 0s;
  }

  &:hover:not(.active) {
    color: var(--oc-color-swatch-brand-hover) !important;
  }

  &:hover,
  &:focus {
    text-decoration: none !important;
  }
  &.active {
    cursor: default;
    overflow: hidden;
  }
}
</style>
