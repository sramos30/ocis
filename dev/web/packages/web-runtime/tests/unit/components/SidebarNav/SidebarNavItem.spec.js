import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import DesignSystem from 'owncloud-design-system'
import GetTextPlugin from 'vue-gettext'

import SidebarNavItem from 'web-runtime/src/components/SidebarNav/SidebarNavItem.vue'
import sidebarNavItemFixtures from '../../../../../../__fixtures__/sidebarNavItems'

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(DesignSystem)
localVue.use(GetTextPlugin, {
  translations: 'does-not-matter.json',
  silent: true
})

const exampleNavItem = sidebarNavItemFixtures[0]

const propsData = {
  name: exampleNavItem.name,
  active: false,
  target: exampleNavItem.route.path,
  icon: exampleNavItem.icon,
  index: 5,
  id: '123'
}

describe('OcSidebarNav', () => {
  it('renders navItem without toolTip if expanded', () => {
    const wrapper = mount(SidebarNavItem, {
      store: new Vuex.Store({}),
      localVue,
      stubs: { 'router-link': true },
      propsData: {
        ...propsData,
        collapsed: false
      }
    })
    expect(wrapper).toMatchSnapshot()
  })

  it('renders navItem with toolTip if collapsed', () => {
    const wrapper = mount(SidebarNavItem, {
      store: new Vuex.Store({}),
      localVue,
      stubs: { 'router-link': true },
      propsData: {
        ...propsData,
        collapsed: true
      }
    })
    expect(wrapper).toMatchSnapshot()
  })
})
