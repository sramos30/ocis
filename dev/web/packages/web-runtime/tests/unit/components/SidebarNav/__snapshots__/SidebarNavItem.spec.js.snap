// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OcSidebarNav renders navItem with toolTip if collapsed 1`] = `
<li class="oc-sidebar-nav-item oc-pb-xs oc-px-s" id="123">
  <router-link-stub to="/files/list/all" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-sidebar-nav-item-link" data-nav-id="5"><span class="oc-flex"><span class="oc-icon oc-icon-m oc-icon-passive"><!----></span> <span class="oc-ml-m text text-invisible">Personal</span></span>
    <!---->
    <!---->
  </router-link-stub>
</li>
`;

exports[`OcSidebarNav renders navItem without toolTip if expanded 1`] = `
<li class="oc-sidebar-nav-item oc-pb-xs oc-px-s" id="123">
  <router-link-stub to="/files/list/all" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-passive oc-button-passive-raw oc-sidebar-nav-item-link" data-nav-id="5"><span class="oc-flex"><span class="oc-icon oc-icon-m oc-icon-passive"><!----></span> <span class="oc-ml-m text">Personal</span></span>
    <!---->
    <!---->
  </router-link-stub>
</li>
`;
